### ISSUE TITLE
Release best RetinaFace Pretrained model?

### ISSUE THREAD

**Issue by cdulhanty** [deepinsight/insightface#847]
Hello, as the WiderFace Challenge (ICCV 2019) has now closed, is it possible to release your best pre-trained RetinaFace model? Thank you!

**Comment by tund-ts** [comment]
Hi @nttstar , is there any chance you can release this soon?

**Comment by cclcalder** [comment]
Is there any update on when this will be released? Or the associated stats?

**Comment by EliasVansteenkiste** [comment]
Is the best model already released?

**Comment by dav-ell** [comment]
Also interested in this.

**Comment by rabysam28** [comment]
Would you please respond to this post? I would love to reproduce [this picture](https://github.com/deepinsight/insightface/blob/master/resources/11513D05.jpg) that you have put in the wiki page, but I do not seem to be able to do that by R50 model. Are you planning to release your best model? Thanks

**Comment by ashuezy** [comment]
@nttstar Any updates ?

**Comment by ashuezy** [comment]
Gentle reminder

**Comment by nttstar** [comment]
The best model has many tricks and is not easy enough to be published.

**Comment by deimsdeutsch** [comment]
@nttstar 
Thanks for the work done in this repo but i am seeing for the last 6-8 months that this repo is not actively maintained. All the models that being released are not production grade but rather leftovers. So i am not being forced to move on. 

Thanks for the work.

**Comment by nttstar** [comment]
@deimsdeutsch The biggest model is not `production-grade`. I don't think you want to serve your customers for a big time delay.

**Comment by dexception** [comment]
@nttstar 
12 Aug 2019, since the first comment and now you reply ? Who are you ? God ? Is your ego hurt by @deimsdeutsch comments. That is why you replied twice. Get down to earth.

**Comment by ashuezy** [comment]
> @deimsdeutsch The biggest model is not `production-grade`. I don't think you want to serve your customers for a big time delay.

Real world feedback.
Don't get me wrong but there are issues with the biggest model. I have used your model in various live demos and 1%-3% FNMR and 2%-4% FMR. In some cases results were so bad people i had to stop the demo midway.

**Comment by nttstar** [comment]
@ashuezy Hi, I can't see your demos but for all one stage detectors you need to care about the input scales(resolution) with anchor settings.

**Comment by ashuezy** [comment]
@nttstar  I have taken care of these settings. Is it possible that you can share the biggest model trained on deepglint dataset ? I would have trained it myself but i don't have that many GPU's on a machine.

**Comment by nttstar** [comment]
@ashuezy We're talking about detection models(retinaface pretrained models), do you misunderstand? `Deepglint` which you just have said is a recognition dataset.

**Comment by ashuezy** [comment]
@nttstar 
I am aware what i wrote. Apologies if i am changing the topic of the thread but for the last 2 years i am stuck with the same model. That is why asking for a better version of the recognition model. You decide what you have in mind, i am sure people would love to see a better version being released. Cheers !

**Comment by nttstar** [comment]
@ashuezy Industrial grade recognition model requires much more training data (such as 1M identities, 10M identities, or even more.). It is very relevant to how you can make money. I believe the combination of ms1m+glint still can not reach your precision requirement.

**Comment by ashuezy** [comment]
@nttstar
Thanks for sharing this information. 

I would look to buy such a dataset if it is available. For the time being i can see that the number of people in deepglint is 2.12 times more. Still i would request if you can share a trained model for the deepglient for the time being. 

MS1M-ArcFace - 85K ids
DeepGlint -181K ids

**Comment by uday60** [comment]
> @ashuezy Industrial grade recognition model requires much more training data (such as 1M identities, 10M identities, or even more.). It is very relevant to how you can make money. I believe the combination of ms1m+glint still can not reach your precision requirement.

Approach 1
========
Step 1.  Generate 1-10 million faces using styleGAN2 like this website
https://www.thispersondoesnotexist.com/
(You might have to filter lot of data to remove child images and remove multiple faces if any)

Step2. Add emotion augmentation, add image effects, age effects, beard effects if male, alter face angles, hair on forehead, add mask, cover partial face, adult->child. 

Step 3. Train

Approach 2 (Which i don't know exist but sounds more feasible)
========
Find a GAN that takes a single image of a face and generates similar faces.

**Comment by woctezuma** [comment]
Hi there,

I would like to see the best "face detection" RetinaFace model shared for the rest of us.

> RetinaFace Pretrained Models
>
> Pretrained Model: RetinaFace-R50 (baidu cloud or dropbox) is a medium size model with ResNet50 backbone.
>
> To avoid the confliction with the WiderFace Challenge (ICCV 2019), we postpone the release time of our best model.

Or would you have some tips (especially scales) to get the most out of the pre-trained model?

**Comment by ashuezy** [comment]
@nttstar 
Any updates regarding the best face detection model being available?
